static/js/components/AddToListDialog.js

```diff
@@ -0,0 +1,271 @@
+                    onChange={(e: any) => {
+                      toggleListItem(resource, userList, !e.target.checked)
+                    }}
+                  >
+                    {`${userList.title}`}
+                  </Checkbox>
+                </div>
+                <div>
+                  <div className="grey-surround privacy">
+                    <i
+                      className={`material-icons ${privacyIcon(
+                        userList.privacy_level
+                      )}`}
+                    >
+                      {privacyIcon(userList.privacy_level)}
+                    </i>
+                    {capitalize(userList.privacy_level)}
+                  </div>
+                </div>
+              </div>
+            ) : null
+      )}
+    </div>
+  )
+  return resource && isFinished ? (
+    <Dialog
+      id="list-add-dialog"
+      open={!emptyOrNil(resource)}
+      hideDialog={hideDialog}
+      title="Add to List"
+      onAccept={hideDialog}
+      hideCancel={true}
+      submitText="OK"
+      className="user-listitem-dialog"
+    >
+      {renderAddToListForm()}
+    </Dialog>
+  ) : null
+}
+
+const getObject = createSelector(
+  state => state.ui,
+  state => state.entities.courses,
+  state => state.entities.bootcamps,
+  state => state.entities.programs,
+  state => state.entities.userLists,
+  state => state.entities.videos,
+  state => state.queries,
+  (ui, courses, bootcamps, programs, userLists, videos, queries) => {
+    const object = ui.dialogs.get(DIALOG_ADD_TO_LIST)
+
+    if (object && object.object_type) {
+      switch (object.object_type) {
+      case LR_TYPE_COURSE:
+        return querySelectors.isFinished(queries, courseRequest(object.id))
+          ? courses[object.id]
+          : null
+      case LR_TYPE_BOOTCAMP:
+        return querySelectors.isFinished(queries, bootcampRequest(object.id))
+          ? bootcamps[object.id]
+          : null
+      case LR_TYPE_PROGRAM:
+        return querySelectors.isFinished(queries, programRequest(object.id))
+          ? programs[object.id]
+          : null
+      case LR_TYPE_VIDEO:
+        return querySelectors.isFinished(queries, videoRequest(object.id))
+          ? videos[object.id]
+          : null
+      case LR_TYPE_USERLIST:
+        return querySelectors.isFinished(queries, userListRequest(object.id))
+          ? userLists[object.id]
+          : null
+      case LR_TYPE_LEARNINGPATH:
+        return querySelectors.isFinished(queries, userListRequest(object.id))
+          ? userLists[object.id]
+          : null
+      }
+    }
+    return null
+  }
+)
+
+export const mapStateToProps = (state: Object) => {
+  const { ui } = state
+  const object = ui.dialogs.get(DIALOG_ADD_TO_LIST)
+  const objectId = object ? object.id : null
+  const objectType = object ? object.object_type : null
+
+  return {
+    objectId,
+    objectType,
+    resource: getObject(state)
+  }
+}
```

Comment:
why not put the `getObject` selector into a `useSelector` hook like the one for userlists?

Reply:
actually, see my comment below for a suggestion about removing the need for the `getObject` and `mapPropsToConfig` for this component entirely